bin/ch/Helpers.cpp

```diff
@@ -16,244 +16,104 @@
 
 //TODO: x-plat definitions
 #ifdef _WIN32
-typedef char16 TTDHostCharType;
-typedef struct _wfinddata_t TTDHostFileInfo;
-typedef intptr_t TTDHostFindHandle;
-typedef struct _stat TTDHostStatType;
+#define MAX_URI_LENGTH 512
+#define TTD_HOST_PATH_SEP "\\"
 
-#define TTDHostPathSeparator _u("\\")
-#define TTDHostPathSeparatorChar _u('\\')
-#define TTDHostFindInvalid -1
-
-size_t TTDHostStringLength(const TTDHostCharType* str)
-{
-    return wcslen(str);
-}
-
-void TTDHostInitEmpty(TTDHostCharType* dst)
-{
-    dst[0] = _u('\0');
-}
-
-void TTDHostInitFromUriBytes(TTDHostCharType* dst, const byte* uriBytes, size_t uriBytesLength)
-{
-    memcpy_s(dst, MAX_PATH * sizeof(TTDHostCharType), uriBytes, uriBytesLength);
-    dst[uriBytesLength / sizeof(TTDHostCharType)] = _u('\0');
-
-    AssertMsg(wcslen(dst) == (uriBytesLength / sizeof(TTDHostCharType)), "We have an null in the uri or our math is wrong somewhere.");
-}
-
-void TTDHostAppend(TTDHostCharType* dst, size_t dstLength, const TTDHostCharType* src)
-{
-    size_t srcLength = TTDHostStringLength(src);
-    size_t dpos = TTDHostStringLength(dst);
-    Helpers::TTReportLastIOErrorAsNeeded(dpos < dstLength, "The end of the string already exceeds the buffer");
-
-    size_t srcByteLength = srcLength * sizeof(TTDHostCharType);
-    size_t dstRemainingByteLength = (dstLength - dpos - 1) * sizeof(TTDHostCharType);
-    Helpers::TTReportLastIOErrorAsNeeded(srcByteLength <= dstRemainingByteLength, "The source string must be able to fit within the destination buffer");
-
-    memcpy_s(dst + dpos, dstRemainingByteLength, src, srcByteLength);
-    dst[dpos + srcLength] = _u('\0');
-}
-
-void TTDHostAppendWChar(TTDHostCharType* dst, size_t dstLength, const wchar* src)
+void TTDHostBuildCurrentExeDirectory(char* path, size_t* pathLength)
 {
-    size_t srcLength = wcslen(src);
-    size_t dpos = TTDHostStringLength(dst);
-    Helpers::TTReportLastIOErrorAsNeeded(dpos < dstLength, "The end of the string already exceeds the buffer");
-
-    size_t dstRemainingLength = dstLength - dpos - 1;
-    Helpers::TTReportLastIOErrorAsNeeded(srcLength <= dstRemainingLength, "The source string must be able to fit within the destination buffer");
+    wchar exePath[MAX_PATH];
+    GetModuleFileName(NULL, exePath, MAX_PATH);
 
-    for(size_t i = 0; i < srcLength; ++i)
+    size_t i = wcslen(exePath) - 1;
+    while(exePath[i] != _u('\\'))
     {
-        dst[dpos + i] = (char16)src[i];
+        --i;
     }
 
-    dst[dpos + srcLength] = _u('\0');
+    if(i * 3 > MAX_URI_LENGTH)
```

Comment:
Is pathLength an out-param or in-out? I would have thought it would be in as in it initially contains the buffer length, and after the call, it's set to the string byte count- seems like that would be a safer pattern here instead of making assumptions?